lib/Sema/TypeCheckSwitchStmt.cpp

```diff
@@ -1086,11 +1118,29 @@ namespace {
         } else if (auto *TTy = tp->castTo<TupleType>()) {
           // Decompose each of the elements into its component type space.
           SmallVector<Space, 4> constElemSpaces;
-          llvm::transform(TTy->getElements(),
-                          std::back_inserter(constElemSpaces),
-                          [&](TupleTypeElt elt) {
-            return Space::forType(elt.getType(), elt.getName());
-          });
+
+          auto elements = TTy->getElements();
+          switch (elements.size()) {
+          case 0:
+            break;
```

Comment:
This seems wrong; a zero-element tuple still needs to be matched. Or is that already guaranteed by the pattern-checking logic that ran before this code?

Reply:
It's indeed is matched before it's a `break` and not a `return`, so we'd get `Space::forConstructor` created for it.

Reply:
Ah! Misread that. Thanks.